packages/firestore/src/local/indexeddb_mutation_queue.ts

```diff
@@ -510,6 +529,19 @@ export class IndexedDbMutationQueue implements MutationQueue {
         }
       }
     }
+    // PORTING NOTE: Multi-tab only.
+    // Keeping track of removed mutation batches allows us to ensure that batch
+    // IDs always advance during the lifetime of all clients.
+    promises.push(
+      this.getMutationQueueMetadata(transaction).next(metadata => {
+        if (maxProcessedBatchId > metadata.lastProcessedBatchId) {
```

Comment:
FWIW, it seems like this should be always true, no?  If so we could consider making it an assert.

Reply:
Oh, yeah.  I think I knew that once.  We should really get rid of held write acks. :)